python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_graph/_digraph_group_chat.py

```diff
@@ -465,9 +465,31 @@ async def select_speaker(self, thread: Sequence[BaseAgentEvent | BaseChatMessage
             # Reset the bookkeeping for the specific activation groups that were triggered
             self._reset_triggered_activation_groups(speaker)
 
-        # If there are no speakers, the graph execution is complete.
-        # Signal termination directly since no agents will be selected.
-        if not speakers:
+        return speakers
+
+    async def validate_group_state(self, messages: List[BaseChatMessage] | None) -> None:
+        pass
+
+    async def _apply_termination_condition(
+        self, delta: Sequence[BaseAgentEvent | BaseChatMessage], increment_turn_count: bool = False
+    ) -> bool:
+        """Apply termination condition including graph-specific completion logic.
+        
+        First checks standard termination conditions, then checks if graph execution is complete.
+        
+        Args:
+            delta: The message delta to check termination conditions against
+            increment_turn_count: Whether to increment the turn count
+            
+        Returns:
+            True if the conversation should be terminated, False otherwise
+        """
+        # First apply the standard termination conditions from the base class
+        if await super()._apply_termination_condition(delta, increment_turn_count):
+            return True
+        
+        # Check if the graph execution is complete (no ready speakers)
+        if not self._ready:
```

Comment:
This termination check should be prioritized before the standard termination conditions. It should also reset the termination conditions. See the implementation in the base class for details.